src/jit/lower.cpp

```diff
@@ -1990,6 +1990,7 @@ GenTree* Lowering::LowerTailCallViaHelper(GenTreeCall* call, GenTree* callTarget
 //
 // Notes:
 //    - Decomposes long comparisons that feed a GT_JTRUE (32 bit specific).
+//    - Decomposes long comparisons that produce a value (X86 specific).
```

Comment:
> Decomposition takes care of nodes that produce a long value, comparisons only consume long values

Yes, but those long values are necessarily decomposed. There is precedent for checking the user of a freshly-decomposed node and then decomposing the user as well; this is how we handle `GT_LONG` nodes that are part of field list arguments.

> I think there's a more general question: do we still need decomposition or all it does can now be done during lowering?

Yeah, this is a question that we need to think through. Off the top of my head it seems like we should at least be able to do
```
foreach block:
    foreach node:
        range newNodes = decomposeNode(node)
        lowerRange(newNodes)
```
rather than today's
```
foreach block:
    decompose_block()
    lower_block()
```